src/panels/utilities/ClusterSpecCreationBuilder.ts

```diff
@@ -0,0 +1,236 @@
+        //TODO - implement this
+        const parameters = {
+            "location": {
+                "value": clusterSpec.location
+            },
+            "resourceName": {
+                "value": clusterSpec.name
+            },
+            "dnsPrefix": {
+                "value": `${clusterSpec.name}-dns`
+            },
+            "apiVersion": {
+                "value": ClusterSpecBuilder.apiVersion
+            },
+            "nodeResourceGroup": {
+                "value": `MC_${clusterSpec.resourceGroupName}_${clusterSpec.name}_${clusterSpec.location}`
+            },
+            "subscriptionId": {
+                "value": clusterSpec.subscriptionId
+            },
+            "resourceGroupName": {
+                "value": clusterSpec.resourceGroupName
+            },
+            "kubernetesVersion": {
+                "value": clusterSpec.kubernetesVersion
+            },
+            "clusterIdentity": {
+                "value": {
+                    "type": "SystemAssigned"
+                }
+            }
+        }
+
+        const deploymentParameters: Deployment = {
+            "properties": {
+                "parameters": parameters,
+                "template": devTestTemplate, //TODO - change this to the correct template
+                "mode": "Incremental"
+            }
+        };
+        return deploymentParameters;
+    }
+
+    buildProdEnterpriseClusterSpec(clusterSpec: ClusterSpec): Deployment {
+        //TODO - implement this
+        const parameters = {
+            "location": {
+                "value": clusterSpec.location
+            },
+            "resourceName": {
+                "value": clusterSpec.name
+            },
+            "dnsPrefix": {
+                "value": `${clusterSpec.name}-dns`
+            },
+            "apiVersion": {
+                "value": ClusterSpecBuilder.apiVersion
+            },
+            "nodeResourceGroup": {
+                "value": `MC_${clusterSpec.resourceGroupName}_${clusterSpec.name}_${clusterSpec.location}`
+            },
+            "subscriptionId": {
+                "value": clusterSpec.subscriptionId
+            },
+            "resourceGroupName": {
+                "value": clusterSpec.resourceGroupName
+            },
+            "kubernetesVersion": {
+                "value": clusterSpec.kubernetesVersion
+            },
+            "clusterIdentity": {
+                "value": {
+                    "type": "SystemAssigned"
+                }
+            },
+        }
+
+        const deploymentParameters: Deployment = {
+            "properties": {
+                "parameters": parameters,
+                "template": devTestTemplate, //TODO - change this to the correct template
+                "mode": "Incremental"
+            }
+        };
+        return deploymentParameters;
+    }
+
+}
+
+function generateDnsPrefix(clusterName: string): string {
```

Comment:
I would avoid calling a validate function that throws an error, because it won't be handled anywhere. We've made a decision not to do explicit validation, and instead allow the deployment to fail if anything is wrong.

We can ensure the DNS prefex adheres to naming rules with a function like this:
```typescript
function generateDnsPrefix(clusterName: string): string {
    return clusterName
        .replaceAll(/[^a-z0-9-\\s]/gi, "")
        .replace(/^-/, "")
        .replace(/-$/, "")
        .substring(0, 54);
}
```